src/main/java/org/example/Extractor.java

```diff
@@ -9,18 +9,58 @@
 import java.util.List;
 
 public class Extractor {
+    
     public List<Chair> getChairs(String url){
-
-        String role, name, email;
         List<Chair> chairs = new ArrayList<>();
-        //todo
+
+        Document document = getDocument(url);
+        if (document == null) return chairs;
+
+        String role = "", name = "", email = "";
```

Comment:
The reason I initialize them here is because they must be guaranteed to have a value when creating a Chair object. _role_ is guaranteed because it gets set to the text on an element unconditionally, but _name_ and _email_ only have values based on certain if branches being hit. The alternative solution to this is giving them values within the for loop like so:

```java
 String role, name, email;

        for (Element tr : document.select("table.table>tbody>tr")) {
            if (tr.children().isEmpty() || !tr.children().first().hasText() || !tr.children().first().text().toLowerCase().contains("chair")) {
                continue;
            }

            role = tr.child(0).text();
            name = "";
            email = "";

            Element facultyDetailsElement = tr.child(1);
            for (Element child : facultyDetailsElement.children()) {
                if (child.tagName().equals("strong")) {
                    name = child.select("a").text();
                } else if (child.tagName().equals("a") && child.attr("abs:href").startsWith("mailto")) {
                    email = child.text();
                }
            }
            chairs.add(new Chair(name, role, email));
        }
```

But this would mean that I am needlessly setting their values at each iteration. Regardless, would you like me to go with that approach or keep the current one?

Reply:
Yes assigning a constant value to a variable in each iteration is not a good practice. assignment should be done outside of the loop.
By the way, I was talking about line 19. I think what you are talking about a different part of the code.

Reply:
Right so I should keep it as is then correct?

Reply:
name and email could be not initiated in the for loop.

Reply:
I made some changes in the latest commit, please let me know if this is what you are looking for.